include/bpf_helper_defs.h

```diff
@@ -339,3 +339,78 @@ EBPF_HELPER(int32_t, bpf_is_current_admin, (const void* ctx));
 #ifndef __doxygen
 #define bpf_is_current_admin ((bpf_is_current_admin_t)BPF_FUNC_is_current_admin)
 #endif
+
+/**
+ * @brief Copy memory from one location to another.
+ *
+ * @param[in] destination Destination buffer.
+ * @param[in] destination_size Size of the destination buffer.
+ * @param[in] source Source buffer.
+ * @param[in] source_size Size of the source buffer.
+ *
+ * @retval 0 The operation was successful.
+ * @retval -EINVAL One or more parameters are invalid.
+ */
+EBPF_HELPER(long, bpf_memcpy, (void* destination, uint32_t destination_size, const void* source, uint32_t source_size));
+#ifndef __doxygen
+#define bpf_memcpy ((bpf_memcpy_t)BPF_FUNC_memcpy)
+#endif
+
+/**
+ * @brief Compare two memory regions.
+ *
+ * @param[in] memory1 First memory region.
+ * @param[in] memory1_size Size of the first memory region.
+ * @param[in] memory2 Second memory region.
+ * @param[in] memory2_size Size of the second memory region.
+ *
+ * @returns 0 if the memory regions are equal, a negative value if memory1 is less than memory2, or a positive value if
+ * memory1 is greater than memory2.
+ */
+
+EBPF_HELPER(int, bpf_memcmp, (const void* memory1, uint32_t memory1_size, const void* memory2, uint32_t memory2_size));
+#ifndef __doxygen
+#define bpf_memcmp ((bpf_memcmp_t)BPF_FUNC_memcmp)
+#endif
+
+/**
+ * @brief Set memory to a specific value.
+ *
+ * @param[in] memory Memory region to set.
+ * @param[in] size Size of the memory region.
+ * @param[in] value Value to set the memory region to.
+ *
+ * @returns Pointer to the memory region, or a negative error in case of failure.
+ */
+
+EBPF_HELPER(long, bpf_memset, (void* memory, uint32_t size, int value));
+#ifndef __doxygen
+#define bpf_memset ((bpf_memset_t)BPF_FUNC_memset)
+#endif
+
+/**
+ * @brief Move memory from one location to another.
```

Comment:
I don't think it actually "moves" the memory (nor does it "move" the _contents_ of memory) which if taken literally would mean it no longer exists in the original location.  Per https://stackoverflow.com/questions/4415910/memcpy-vs-memmove, memmove copies the memory, allowing for overlapping regions, and is a bit slower than memcpy which doesn't safely allow for overlapping regions.  See also https://learn.microsoft.com/en-us/cpp/c-runtime-library/buffer-manipulation?view=msvc-170 which defines memmove as "Copy specified number of characters from one buffer to another" and also has the note "When the source and target areas overlap, only memmove is guaranteed to copy the full source properly."